### PR TITLE
Redisearch indexing

### PR DESCRIPTION
**Pull request by cheeseandcereal**
## Description

This PR swaps out the elasticsearch indexing for redisearch; a lighter weight indexing solution

This will be a breaking change for the query endpoints

##### Checklist

- [x] `./tools.sh full-test` passes
- [x] tests are included
- [x] documentation is changed or added
- [x] changelog has been modified for the changes

### CODE REVIEW

File: dragonchain/lib/dao/smart_contract_dao.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 def get_contract_by_txn_type(txn_type: str) -> "smart_contract_model.SmartContractModel":
     """Searches for a contract by txn_type"""
-    results = redisearch.search(index_name=FOLDER, query=txn_type)["results"]
+    results = redisearch.search(index="sc", query_str=f"@txn_type:{txn_type}")["results"]
```
Comment (cheeseandcereal) [inline dragonchain/lib/dao/smart_contract_dao.py]: this wont' work. Search used to pull from S3. We'll need to do that manually now. Also I don't even think ['results'] is correct on a redisearch result

File: dragonchain/webserver/lib/blocks.py [hunk 2]
```diff
@@ -59,6 +60,10 @@ def query_blocks_v1(params: Optional[dict], parse: bool) -> "RSearch":
         return redisearch.search(index="bk", query_str="*", sort_by="block_id", sort_asc=False)
 
 
+def list_blocks_v1(get_whole_object: bool) -> List[str]:
+    return storage.list_objects("BLOCK")
```
Comment (cheeseandcereal) [inline dragonchain/webserver/lib/blocks.py]: We definitely can't do this, there are verifications blocks in there too for L1's. They'll need to be filtered out

File: dragonchain/webserver/lib/transactions.py [hunk 3]
```diff
@@ -37,6 +37,49 @@
...
+        query_params["sort"] = sort
+    if offset:
+        query_params["offset"] = offset
+    if limit:
+        query_params["limit"] = limit
+    hits_pages = redisearch.search(index=transaction_dao.FOLDER, query_str=query_str, **query_params)
+    _log.info(f"pages: {hits_pages}")
+    storage_objects = []
+
+    for hit in hits_pages["hits"]:
```
Comment (cheeseandcereal) [inline dragonchain/webserver/lib/transactions.py]: this is definitely not how redisearches work/return

File: dragonchain/webserver/routes/blocks.py [hunk 4]
```diff
@@ -25,20 +25,25 @@
...
+    app.add_url_rule("/v1/block", "list_or_query_blocks_v1", list_or_query_blocks_v1, methods=["GET"])
     app.add_url_rule("/block/<block_id>", "get_block_v1", get_block_v1, methods=["GET"])
     app.add_url_rule("/v1/block/<block_id>", "get_block_v1", get_block_v1, methods=["GET"])
 
 
 @request_authorizer.Authenticated()
 def get_block_v1(block_id: str) -> Tuple[str, int, Dict[str, str]]:
-    should_parse = bool(flask.request.headers.get("Parse-Payload"))
-    return helpers.flask_http_response(200, blocks.get_block_by_id_v1(block_id, should_parse))
+    get_whole_objects = bool(flask.request.headers.get("Get-Objects")) or True
```
Comment (rdaquilante) [inline dragonchain/webserver/routes/blocks.py]: restore should_parse

File: dragonchain/webserver/lib/smart_contracts.py [hunk 5]
```diff
@@ -76,22 +58,25 @@ def create_contract_v1(body: dict) -> dict:
         DTO of the contract at rest which is being created
     """
     # Before anything else, check to see if this chain has too many contracts
-    if elasticsearch.get_count(folder=smart_contract_dao.FOLDER) >= MAX_CONTRACT_LIMIT:
+    if len(storage.list_objects(smart_contract_dao.FOLDER)) >= MAX_CONTRACT_LIMIT:
```
Comment (cheeseandcereal) [inline dragonchain/webserver/lib/smart_contracts.py]: can't do this

**Comment by cheeseandcereal** [comment]
Merge on this is going to wait for #176 to happen first

**Pull request merged**